### DISCUSSION TITLE
Takeover firebase_functions_interop

### DISCUSSION THREAD

**Discussion by stargazing-dino** [firebase/flutterfire#4327] [Feature request]
Hi!

I think it would be nice if the awesome team here could consider creating and managing a library similar to [firebase_functions_interop](https://pub.dev/packages/firebase_functions_interop). The current maintainer, @pulyaevskiy, has priorities in [zefyr](https://github.com/memspace/zefyr) so I doubt he'll have time for another project of that scale.

The benefits of writing functions in Dart are numerous, including using the same models on server as well as front end. I also think there would be a lot of interest in it if it gained official support. This [Medium article](https://medium.com/icnh/writing-cloud-functions-in-dart-b7e62192b3bc) explaining the setup of firebase_functions_interop  for example has ~800 claps.

"Why not just wait for Firebase to support Dart Cloud Functions?" According to [this issue](https://github.com/firebase/firebase-functions/issues/383) they have no near term plans to add support for any other languages although the initial proposal did have 44 upvotes.

I also think this potential package would be a good fit here because with the expanding web support there are a lot of experienced contributors here who are great with js-dart interop.

**Comment by pulyaevskiy** [comment]
Just want to point out that there is actually some work on adding support for cloud functions with Dart

https://github.com/GoogleCloudPlatform/functions-framework-dart

Since firebase functions are just cloud functions with some sugar it shouldn’t be that hard to add support for Firebase...

Though I’m not involved in that project and do not work at Google to have any insights about the state of it.

@subfuzion may be able to provide more details.

**Comment by Salakar** [comment]
I was snooping around `firebase_functions_interop` last night actually, was curious. I may be interested in helping maintain it; however it might only be from a personal perspective rather than Invertase, not sure yet.

**Reply by pulyaevskiy** [reply]
Feel free to reach out to me if you're interested in contributing to the interop package.
